File: common/src/main/java/com/syc/common/widget/refresh/adapter/BaseViewHolder.java
package com.syc.common.widget.refresh.adapter;

import android.databinding.ViewDataBinding;
import android.support.v7.widget.RecyclerView;
import android.view.View;

/**
 * Created by shiyucheng on 2018/7/7.
 */

public class BaseViewHolder extends RecyclerView.ViewHolder {
    private ViewDataBinding binding;
    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    public BaseViewHolder(ViewDataBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public <T extends ViewDataBinding> T getBinding(){
        return (T) binding;
    }

}

File: acount/src/main/kotlin/com/syc/acount/view/adapter/BaseViewHolder.kt
package com.syc.framework.sycframework.view.adapter

import android.databinding.ViewDataBinding
import android.support.v7.widget.RecyclerView

/**
 * Created by shiyucheng on 2018/1/11.
 */
class BaseViewHolder<T : ViewDataBinding>(var binding: T) : RecyclerView.ViewHolder(binding.root)